### PR TITLE
[Access Requests] Support SSH node resource constraints in Web UI

### PR DESCRIPTION
**Pull request by kiosion**
## Context
Web UI support for SSH node Resource Constraints, building on the existing AWS Console constraint UI added with https://github.com/gravitational/teleport/pull/60123. Allows users to select specific SSH logins when adding nodes to an Access Request, and displays SSH login constraints during request review.

### Changes
- Add `SshConstraints` type and `toggleSSHConstraint` util for managing SSH login selections
- Add `SSHConstraintsList` component to the request checkout flow and the request review flow
- Add `SSHLogin` type and `sshLoginDetails` field to node type for per-login requestable details
- Add `supportedFeatureIds` to `UnifiedResourceNode` for feature gating
- Add stories for SSH constraint checkout flow

changelog: Add Web UI support for selecting SSH logins as Resource Constraints when creating Access Requests for SSH node resources.

## Testing
Added/updated:
- `RequestCheckout.test.tsx`: Renders SSH constraint logins and AWS Console constraint ARNs in checkout
- `utils.test.ts`: `toggleSSHConstraint` and `toggleAWSConsoleConstraint` toggle logic
- `makeNode.test.ts`: Node resource parsing with `sshLoginDetails`, edge cases

## Test Plan
- [x] Request Checkout:
  - [x] New request can be created for an SSH node, with specific login(s) selected as constraints
  - [x] SSH login list visible for SSH node with constraints applied
  - [x] Nodes without constraints render normally alongside constrained nodes
  - [x] Request can be submitted, reviewed, and approved/denied for:
    - [x] Request containing single constrained SSH node resource
    - [x] Request containing mixed constrained/unconstrained SSH node resources
    - [x] Request containing both constrained AWS Console resource and SSH node resource
- [x] Existing tested AWS Console constraint flows are unaffected
- [x] TSH may be used to review/assume an Access Request containing constrained SSH node resources created via Web UI

## Screenshots
- One login granted: 
Screenshot 2026-03-12 at 19 10 07

- One login granted, one requestable: 
Screenshot 2026-03-12 at 19 10 13

- One login requestable: 
Screenshot 2026-03-12 at 19 10 19

## Related
- Requires https://github.com/gravitational/teleport/pull/64403
- Required for https://github.com/gravitational/teleport.e/pull/8200
- See https://github.com/gravitational/teleport/issues/58307

### CODE REVIEW

File: web/packages/teleport/src/services/nodes/makeNode.ts [hunk 1]
```diff
@@ -43,11 +45,32 @@ export default function makeNode(json: any): Node {
...
+  );
+}
+
+function makeSSHLoginDetails(raw: any): SSHLogin[] | undefined {
+  if (!Array.isArray(raw)) {
+    return undefined;
+  }
+  return raw.map((item: any) => ({
+    login: item.login ?? '',
+    requiresRequest: item.requiresRequest,
```
[the remark this answers is not in the document]
Comment (kiosion) [reply web/packages/teleport/src/services/nodes/makeNode.ts]: ctx: `sshLoginDetails` arrays from `json.Marshal` won't contain null entries, Go serializes `[]SSHLogin` as an array of objs. I think we can remove this parsing entirely though; the serialized SSHLogin shape is identical.

**Comment by ryanclark** [comment]
Can you add some screenshots of what the UI changes look like?

**Comment by kiosion** [comment]
@ryanclark added a screenshot for each possible state, LMK if "before" vs "after" would be better, can also do that instead!

**Review by ryanclark — APPROVED** [review]

**Review by charlestp — APPROVED** [review]
I was looking at that `Request Access` button spacing in your screenshot, outside this PR, and looks like you already have a todo for it 😄
